azurerm/internal/services/storage/storage_account_resource.go

```diff
@@ -1007,10 +1092,9 @@ func resourceStorageAccountUpdate(d *schema.ResourceData, meta interface{}) erro
 	if d.HasChange("blob_properties") {
 		// FileStorage does not support blob settings
 		if accountKind != string(storage.FileStorage) {
-			blobClient := meta.(*clients.Client).Storage.BlobServicesClient
 			blobProperties := expandBlobProperties(d.Get("blob_properties").([]interface{}))
 
-			if _, err = blobClient.SetServiceProperties(ctx, resourceGroupName, storageAccountName, blobProperties); err != nil {
+			if _, err = accountsClient.SetServiceProperties(ctx, storageAccountName, blobProperties); err != nil {
```

Comment:
What is the reason we are changing the client used here?

Reply:
`accountsClient` is a `BlobServicePropertiesClient` whereas `blobClient` was a `BlobServicesClient`, which does not contain the new fields managed in this PR.